src/extension.ts

```diff
@@ -20,62 +20,322 @@ import * as vscode from 'vscode';
+	context.subscriptions.push(startIntegration);
 
 	// populate the initial tree
 	camelKIntegrationsProvider.refresh();
 }
 
-// "run" the integration file with "kamel"
-function runTheFile(context: vscode.ExtensionContext) {
+// find an output channel by name in the map and remove it
+function removeOutputChannel( channelName: string) {
+	let podOutputChannel;
+	if (outputChannelMap.has(channelName)) {
+		podOutputChannel = outputChannelMap.get(channelName);
+		if (podOutputChannel !== undefined) {
+			podOutputChannel.dispose();
+			outputChannelMap.delete(channelName);
+			utils.shareMessage(mainOutputChannel, `Removed Output channel for integration: ${channelName}`);
+			mainOutputChannel.show();
+		}
+	}
+}
+
+// retrieve the actual output channel for the name we stashed in the map
+function getOutputChannel( channelName: string) : vscode.OutputChannel {
+	let podOutputChannel;
+	if (outputChannelMap.has(channelName)) {
+		podOutputChannel = outputChannelMap.get(channelName);
+		if (podOutputChannel !== undefined) {
+			return podOutputChannel;
+		}
+	}
+	podOutputChannel = vscode.window.createOutputChannel(channelName);
+	outputChannelMap.set(channelName, podOutputChannel);
+	return podOutputChannel;
+}
+
+// start the integration file
+async function runTheFile(context: vscode.ExtensionContext) {
 	startIntegration(context);
-	camelKIntegrationsProvider.refresh();
+	await camelKIntegrationsProvider.refresh();
 }
 
-// "stop" a running integration in the camel-k system
-function stopTheFile(context: vscode.ExtensionContext) {
-	stopIntegration(context);
-	camelKIntegrationsProvider.refresh();
+// locate the POD name that corresponds to the Integration name
+function findThePODNameForIntegrationFromJSON(json : Object, integrationName : string) : Promise<string> {
+	return new Promise( async (resolve, reject) => {
+		if (json) {
+			let temp = JSON.stringify(json);
+			let o = JSON.parse(temp);
+			for (var i=0; i<o.items.length;i++) {
+				var metadataName : string = o.items[i].metadata.name;
+				if (metadataName.startsWith(integrationName + '-')) {
+					resolve(metadataName);
+				}
+			}
+			reject(new Error());
+		}
+	});
 }
 
-// call the kamel utility to start an integration from a file
+// use curl to do a rest call to follow the logs and process output to the output channel
+function getPodLogViaCurl(podName : string): Promise<boolean> {
+	return new Promise( async (resolve, reject) => {
+			let podsURL = utils.createCamelKPodLogURL(podName);
+			await utils.pingTheURL(podsURL).catch( (error) =>  {
+				reject(error);
+			});
+			let podOutputChannel = getOutputChannel(podName);
+			podOutputChannel.show();
+			let commandString = `curl -v -H "Accept: application/json, */*" ${podsURL}?follow=true --insecure -N`;
+			await utils.delay(1000);
+			let runKamel = child_process.spawn(curlCommand, [curlOption, commandString]);
+			runKamel.stdout.on('data', function (data) {
+				podOutputChannel.append(`${data}`);
```

Comment:
maybe in a next iteration but it seems that Kubernetes extension is providing a nice Log viewer as a view or an editor. it might be better than using the podOutputChannel (and still cannot test but it seems that in case of several pod, the log will be mixed in it, isn't it?)

Reply:
So far as I know, we can't rely on elements from other extensions -- has that changed?

Reply:
you're right, I tend to forget that VS Code is so powerless...
Then maybe in a next iteration, duplicate the log editor code (or find a way to extract it as a normal javascript extension and submit it)

Reply:
Created https://github.com/camel-tooling/vscode-camelk/issues/39